docs/source/guides/data.rst

```diff
@@ -0,0 +1,548 @@
+
+.. testsetup:: *
+
+    from pytorch_lightning.core.Lightning import LightningModule
+    from torch.utils.data import IterableDataSet
+    from pytorch_lightning.trainer.trainer.Trainer import Trainer
+
+.. _data:
+
+#############
+Managing Data
+#############
+
+Continue reading to learn about:
+
+* `Data Objects in Lightning <Data Objects in Lightning_>`_
+
+* `Models Defined by Data <Models Defined by Data_>`_
+
+* `Multiple Datasets <Multiple DataSets_>`_
+
+* `Sequential Data <Sequential Data_>`_
+
+*************************
+Data Objects in Lightning
+*************************
+
+There are a few different Data objects used in Lightning:
+
+.. list-table:: Data objects
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Object
+     - Definition
+   * - :class:`~torch.utils.data.Dataset`
+     - The PyTorch :class:`~torch.utils.data.Dataset` represents a map from keys to data samples.
+   * - :class:`~torch.utils.data.DataLoader`
+     - The PyTorch :class:`~torch.utils.data.DataLoader` represents a Python iterable over a DataSet.
+   * - :class:`~pytorch_lightning.core.datamodule.LightningDataModule`
+     - A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of a training DataLoader, validation DataLoader(s) and test DataLoader(s), along with the matching transforms and data processing/downloads steps required.
+
+Creating DataLoaders
+====================
+
+To pass in data to Lightning, you'll first need to create a :class:`~torch.utils.data.DataLoader` from a :class:`~torch.utils.data.Dataset`.
+
+For example, here's the PyTorch code for loading an MNIST :class:`~torch.utils.data.DataLoader`:
+
+.. testcode::
+    :skipif: not _TORCHVISION_AVAILABLE
+
+    from torch.utils.data import DataLoader, random_split
+    from torchvision.datasets import MNIST
+    import os
+    from torchvision import datasets, transforms
+
+    # transforms
+    # prepare transforms standard to MNIST
+    transform=transforms.Compose([transforms.ToTensor(),
+                                  transforms.Normalize((0.1307,), (0.3081,))])
+
+    # data
+    mnist_train = MNIST(os.getcwd(), train=True, download=True, transform=transform)
```

Comment:
Should this really be `os.getcwd` and potentially pollute the users current dir or should we use `tempfile.mkdtemp()` instead?